branchpro/posterior.py

```diff
@@ -0,0 +1,199 @@
+#
+# BranchProPosterior Class
+#
+# This file is part of BRANCHPRO
+# (https://github.com/SABS-R3-Epidemiology/branchpro.git) which is released
+# under the BSD 3-clause license. See accompanying LICENSE.md for copyright
+# notice and full license details.
+#
+
+import numpy as np
+import scipy.stats
+import pandas as pd
+
+
+class BranchProPosterior(object):
+    """BranchProPosterior Class:
+    Class for computing the posterior distribution used for the inference of
+    the reproduction numbers of an epidemic in the case of a branching process.
+
+    Choice of prior distribution is the conjugate prior for the likelihood
+    (Poisson) of observing given incidence data, hence is a Gamma distribution.
+    We express it in the shape-rate configuration.
```

Comment:
I think it's helpful to spell out the parameterization, so you can see what form it takes even if you don't remember the terms "rate" vs "scale." Not to mention there are many cases where these terms are used inconsistently. 

```suggestion
    We express it in the shape-rate configuration, so that the PDF takes the
    form:

    .. math::
        f(x) = \frac{\beta^\alpha}{\Gamma(\alpha)} x^{\alpha-1} e^{-\beta x}
```